src/main/java/org/p2p/solanaj/programs/JupiterSwapProgram.java

```diff
@@ -0,0 +1,110 @@
+        } else {
+            throw new IOException("Unexpected response status: " + status);
+        }
+    };
+
+    public static URI createQuoteUri(String inputToken, String outputToken, String amount, String slippage) {
+        try {
+            return new URIBuilder("https://quote-api.jup.ag/v6/quote")
+                    .addParameter("inputMint", inputToken)
+                    .addParameter("outputMint", outputToken)
+                    .addParameter("amount", amount)
+                    .addParameter("slippageBps", slippage)
+                    .build();
+        } catch (URISyntaxException e) {
+            throw new RuntimeException("Failed to create quote URI: ", e);
+        }
+    }
+
+    public static JsonNode getJupiterQuote(URI quoteUri) {
+        HttpGet quoteRequest = new HttpGet(quoteUri);
+
+        JsonNode quote;
+        try {
+            quote = httpClient.execute(quoteRequest, handler);
+        } catch (IOException e) {
+            throw new RuntimeException("Failed to retrieve quote from jupiter for " + quoteUri, e);
+        }
+        return quote;
+    }
+
+    public static String swapToken(RpcClient rpcClient, Account account, JsonNode quote) {
+        SwapRequest swapRequestBody = new SwapRequest(quote, account.getPublicKey().toString());
+        String jsonSwapRequestBody;
+        try {
+            jsonSwapRequestBody = objectMapper.writeValueAsString(swapRequestBody);
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException(e);
+        }
+
+        HttpPost swapRequest = new HttpPost("https://quote-api.jup.ag/v6/swap");
+        swapRequest.setHeader("Content-Type", "application/json");
+        swapRequest.setEntity(new StringEntity(jsonSwapRequestBody));
+
+        JsonNode swapRes;
+        try {
+            swapRes = httpClient.execute(swapRequest, handler);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+
+        String swapTransaction = swapRes.get("swapTransaction").asText();
+
+        byte[] base64Decoded = Base64.getDecoder().decode(swapTransaction);
+        Transaction transaction = Transaction.deserialize(base64Decoded);
+
+        try {
+            return rpcClient.getApi().sendTransaction(transaction, account);
+        } catch (RpcException e) {
+            throw new RuntimeException("Failed to send swap transaction", e);
+        }
+    }
+
+    public static String swapToken(RpcClient rpcClient, Account account, String inputToken, String outputToken, String amount, String slippage) {
+        URI quoteUri = createQuoteUri(inputToken, outputToken, amount, slippage);
+        return swapToken(rpcClient, account, quoteUri);
+    }
+
+    public static String swapToken(RpcClient rpcClient, Account account, URI quoteUri) {
+        JsonNode quote = getJupiterQuote(quoteUri);
+        return swapToken(rpcClient, account, quote);
```

Comment:
Why is this recursion here?

Reply:
That is actually calling a different method. I have changed the local variable name to be not confusing.